twenty-first/src/math/polynomial.rs

```diff
@@ -1177,48 +1254,89 @@ where
     ///
     /// This method uses NTT-based multiplication, meaning that the unstructured
     /// part of the structured multiple must be given in NTT-domain.
-    fn reduce_by_ntt_friendly_modulus(&self, shift_ntt: &[FF], m: usize) -> Self {
+    ///
+    /// This function is marked `pub` for benchmarking. Not considered part of
+    /// the public API
+    #[doc(hidden)]
+    pub fn reduce_by_ntt_friendly_modulus(&self, shift_ntt: &[FF], tail_length: usize) -> Self {
+        // m = tail_length
         let domain_length = shift_ntt.len();
         assert!(domain_length.is_power_of_two());
         let log_domain_length = domain_length.ilog2();
         let omega = BFieldElement::primitive_root_of_unity(domain_length as u64).unwrap();
-        let n = domain_length - m;
+        let n = domain_length - tail_length;
 
-        if self.coefficients.len() < n + m {
+        if self.coefficients.len() < n + tail_length {
             return self.clone();
         }
-        let num_reducible_chunks = (self.coefficients.len() - (m + n) + n - 1) / n;
+        let num_reducible_chunks = (self.coefficients.len() - (tail_length + n) + n - 1) / n;
```

Comment:
Based on the line

```rust
let n = domain_length - tail_length;
```

it seems that every occurence of `n + tail_length` can well be substituted by `domain_length`, which might increase readability. I'm not sure of that, though; if `n + tail_length` provides greater clarity, the following suggestion (which also applies a few more times below) can be ignored.

```suggestion
        let num_reducible_chunks = (self.coefficients.len() - domain_length + n - 1) / n;
```

Additionally, it feels like you are computing the number of chunks somewhat manually here. Would `.div_ceil()` make sense?